Added Kotlin Cheat Sheet

AdamMc331 opened the pull request:
Includes classes/functions/variables/control flow. Would love to add a section in the future about destructuring, delegates, and some other helpful tricks but wanted to get out a starting point.

MclaughlinSteve commented on kotlin.md:
````diff
@@ -0,0 +1,385 @@
+val containsNoEvens = numList.all { it % 2 == 1 }
+val firstEvenNumber: Int = numList.first { it % 2 == 0 }
+val firstEvenOrNull: Int? = numList.firstOrNull { it % 2 == 0 }
+```
````
Consider adding a small section just demonstrating what `it` is

AdamMc331 commented:
Rebased and squashed my commits for this PR.. I don't plan on making any more immediate changes unless they're requested by someone. :) 

@rstacruz please let me know if there's anything else I need to add to this!

asad-awadia commented on kotlin.md:
```diff
@@ -0,0 +1,400 @@
+}
+
+// Above can be replaced with default params
+class Person(val name: String, val age: Int? = null)
```
Would be nice to have a section on init

AdamMc331 commented:
@rstacruz Bumping this one. Let me know what you think. :)

chadwithuhc commented:
Thanks @AdamMc331 !!
